flocker/control/test/test_persistence.py

```diff
@@ -0,0 +1,89 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Tests for ``flocker.control._persistence``.
+"""
+
+from twisted.internet import reactor
+from twisted.trial.unittest import TestCase
+from twisted.python.filepath import FilePath
+
+from .._persistence import ConfigurationPersistenceService
+from .._model import Deployment, Application, DockerImage, Node
+
+
+TEST_DEPLOYMENT = Deployment(nodes=frozenset([
+    Node(hostname=u'node1.example.com',
+         applications=frozenset([
+             Application(
+                 name=u'myapp',
+                 image=DockerImage.from_string(u'postgresql'))]))
+]))
+
+
+class ConfigurationPersistenceServiceTests(TestCase):
+    """
+    Tests for ``ConfigurationPersistenceService``.
+    """
+    def service(self, path):
+        """
+        Start a service, schedule its stop.
+
+        :param FilePath path: Where to store data.
+
+        :return: Started ``ConfigurationPersistenceService``.
+        """
+        service = ConfigurationPersistenceService(reactor, path)
+        service.startService()
+        self.addCleanup(service.stopService)
+        return service
+
+    def test_empty_on_start(self):
+        """
+        If no configuration was previously saved, starting a service results
+        in an empty ``Deployment``.
+        """
+        service = self.service(FilePath(self.mktemp()))
+        self.assertEqual(service.get(), Deployment(nodes=frozenset()))
+
+    def test_directory_is_created(self):
+        """
+        If a directory does not exist in given path, it is created.
+        """
+        path = FilePath(self.mktemp())
+        self.service(path)
+        self.assertTrue(path.isdir())
+
+    def test_file_is_created(self):
+        """
+        If no configuration file exists in the given path, it is created.
+        """
+        path = FilePath(self.mktemp())
+        self.service(path)
+        self.assertTrue(path.child(b"current_configuration.json").exists())
+
+    def test_save_then_get(self):
+        """
+        A configuration that was saved can subsequently retrieved.
+        """
+        service = self.service(FilePath(self.mktemp()))
+        d = service.save(TEST_DEPLOYMENT)
+        d.addCallback(lambda _: service.get())
+        d.addCallback(self.assertEqual, TEST_DEPLOYMENT)
+        return d
+
+    def test_persist_across_restarts(self):
+        """
+        A configuration that was saved can be loaded from a new service.
```

Comment:
Hm.  This creates a second service instance with the same path so I guess it exercises the case where things exist already.